dataplane/linux/bpf_ep_mgr.go

```diff
@@ -718,17 +780,39 @@ func (m *bpfEndpointManager) attachWorkloadProgram(ifaceName string, endpoint *p
 		// We don't have a program attached to this interface yet, attach one now.
 		err := ap.AttachProgram()
 		if err != nil {
-			return err
+			return 0, err
 		}
 
 		jumpMapFD, err = FindJumpMap(ap)
 		if err != nil {
-			return fmt.Errorf("failed to look up jump map: %w", err)
+			return 0, fmt.Errorf("failed to look up jump map: %w", err)
 		}
-		m.setJumpMapFD(ifaceName, polDirection, jumpMapFD)
+		m.setJumpMapFD(ap.Iface, polDirection, jumpMapFD)
 	}
 
-	return m.updatePolicyProgram(jumpMapFD, rules)
+	return jumpMapFD, nil
+}
+
+func (m *bpfEndpointManager) addHostPolicy(rules *polprog.Rules, hostEndpoint *proto.HostEndpoint, polDirection PolDirection) {
+
+	// When there is applicable pre-DNAT policy that does not explicitly Allow or Deny traffic,
+	// we continue on to subsequent tiers and normal or AoF policy.
+	if len(hostEndpoint.PreDnatTiers) == 1 {
+		rules.HostPreDnatTiers = m.extractTiers(hostEndpoint.PreDnatTiers[0], polDirection, NoEndTierDrop)
```

Comment:
Would be slightly nicer to put the len check inside extractTiers.

Also I note you're using a subtly different condition here: `len() == 1` vs `len() > 0` above.